components/builder-scheduler/src/data_store.rs

```diff
@@ -259,15 +291,19 @@ impl DataStore {
         let mut rng = thread_rng();
         let id = rng.gen::<u64>();
 
-        conn.execute("SELECT insert_group_v1($1, $2)",
-                     &[&(id as i64), &project_names])
+        let (project_names, project_idents): (Vec<String>, Vec<String>) =
+            project_tuples.iter().cloned().unzip();
+
+        conn.execute("SELECT insert_group_v1($1, $2, $3)",
```

Comment:
I think the question here is if you are manipulating more than one table within insert_group_v1, you might need to run it from a transaction in order to roll back on failure.